src/zhinst/toolkit/driver/nodes/awg.py

```diff
@@ -249,22 +249,20 @@ def read_from_waveform_memory(self, indexes: t.List[int] = None) -> Waveforms:
         nodes = [
             self.waveform.node_info.path + f"/waves/{index}"
             for index in range(len(waveform_info))
-            if indexes is None or index in indexes
+            if (indexes is None or index in indexes)
+            and int(waveform_info[index]["play_config"])
```

Comment:
maybe add a comment to explain that, it's not clear unless you look at the wfm table

Reply:
Good point.